### PR TITLE
__ids and __count for concept-publisher

### PR DESCRIPTION
**Pull request by kovacshuni**

### CODE REVIEW

**Comment by nwrigh** [comment]
I cannot see any tests for the three new methods

File: handlers.go [hunk 1]
```diff
@@ -17,8 +18,46 @@ func newSpecialReportsHandler(service specialReportService) specialReportsHandle
...
+		log.Warnf("Couldn't write count to HTTP response. count=%d %v\n", len(ids), err)
+		writer.WriteHeader(http.StatusInternalServerError)
+	}
+}
+
+func (h *specialReportsHandler) ids(writer http.ResponseWriter, req *http.Request) {
+	ids := h.service.getSpecialReportIds()
+	writer.Header().Add("Content-Type", "text/plain")
+	if len(ids) == 0 {
+		writer.WriteHeader(http.StatusOK)
```
Comment (tamas-molnar) [inline handlers.go:34]: This seems deceptive, can't we return an empty array? Or a different http status code?

**Review by tamas-molnar — CHANGES_REQUESTED** [review]

File: handlers.go (hunk 1 above)
Comment (tosan88) [reply handlers.go:34]: Yeah, I would prefer a 404 here...

**Review by tosan88 — CHANGES_REQUESTED** [review]

File: handlers.go (hunk 1 above)
Comment (kovacshuni) [reply handlers.go:34]: I also found that the design of concept loading could be enhanced a lot. That's true. I found lots of copy paste and other quickly written code. Not just in this app, but in the other transformers and concept-publisher itself.

Here's an example of another transformer, how it returns ids. It's a list of json entries but not wrapped in a []. 
https://github.com/Financial-Times/curated-authors-transformer#ids

concept-publisher also gets the count.
https://github.com/Financial-Times/concept-publisher/blob/master/service.go#L121

concept-publisher works this way with __ids
https://github.com/Financial-Times/concept-publisher/blob/master/service.go#L184
check line 202, what happens if it's a 404

My task is very high priority so i'm going to make the least changes to fix production.

Comment (tamas-molnar) [reply handlers.go:34]: Ok, fair enough.

**Review by tamas-molnar — APPROVED** [review]

File: handlers.go (hunk 1 above)
Comment (tosan88) [reply handlers.go:34]: Well I'd expect that an empty response for getting the ids is a failure, and I'd expect concept-publisher to report it as a failure.
But as this is high priority, you could refine it with a later PR

**Review by tosan88 — APPROVED** [review]

**Comment by kovacshuni** [comment]
@nwrigh There is a function that's serving, after a few layers, as the healthcheck for this app.
It's all commented out and `return nil`
https://github.com/Financial-Times/special-reports-transformer/blob/master/service.go#L79
Should i test that as well?

What i mean is, this app is good with its tests as it is. I'm hurried by others to release my changes and free up the problem of not being able to publish special reports. I'll try to refactor some things in my limited amount of time, but that's it. I'm sorry.

File: handlers.go (hunk 1 above)
Comment (Nesze) [reply handlers.go:34]: __ids and __count goes hand in hand, if __count returns 0, for me __ids returning an empty response does not suggest a failure. So go ahead.

Note: the type of the body is basically a json stream, which by convention is just `application/json`. That could be updated.

**Review by Nesze — APPROVED** [review]

**Comment by kovacshuni** [comment]
Added tests to the three new methods.

**Pull request merged**